cfme/tests/configure/test_default_views_infra.py

```diff
@@ -41,6 +42,25 @@ def set_and_test_default_view(group_name, view, page):
     assert tb.is_active(view), "{} view setting failed".format(view)
     DefaultView.set_default_view(group_name, old_default)
 
+
+def check_vm_visibility():
+    view = navigate_to(Vm, 'All')
+    value = view.sidebar.vmstemplates.tree.read_contents()
+    # Selecting the path and last Vm in the accordion to perform click
+    tree_root = value[0]
+    tree_provider = value[1][0][0]
+    tree_datacenter = value[1][0][1][0][0]
+    length = len(value[1][0][1][0][1])
+    tree_vm = value[1][0][1][0][1][length - 1]
+    view.sidebar.vmstemplates.tree.click_path(tree_root,
+                                              tree_provider, tree_datacenter, tree_vm)
+    vm = Vm(name=tree_vm, provider=tree_provider)
+    view = vm.create_view(InfraVmDetailsView)
+    try:
+        return view.is_displayed
```

Comment:
Can I use `return "Instance" in view.title.text` here, as getting the provider name to initialize the provider object is not possible always. @mshriver @izapolsk

Reply:
Its unreliable - you could have an appliance with two providers having vms with the same names.  The provider+name combination is what gives you a reliable check.

This does seem a bit unnecessary given the scope and intent of the method. All you really want to test or check here is whether or not the VMs are displayed in the accordion tree - and you can detect that way before trying to click any of them and without creating any additional views.  After calling  a `read_contents` on the tree you should be able to tell given the depth of the content list.

Reply:
Yes, will make changes to accommodate your suggestions 👍